vis.c

```diff
@@ -1578,8 +1578,7 @@ void vis_insert_nl(Vis *vis) {
 	vis_window_invalidate(win);
 }
 
-Regex *vis_regex(Vis *vis, const char *pattern) {
-	size_t len = 0;
+Regex *vis_regex(Vis *vis, const char *pattern, size_t len) {
 	if (!pattern && !(pattern = register_get(vis, &vis->registers[VIS_REG_SEARCH], &len)))
```

Comment:
I'm a little confused  by this. Why does `vis_regex` need to take a length? It just gets overwritten by `register_get` right here. I think most of the changes in the second commit are cascading from this one even though it doesn't seem to do anything.

Reply:
Sorry, I somehow missed that you commented here. Thanks for having a look!

`vis_regex` can either be called with a new regex (`vis_regex(vis, "foo.*", sizeof("foo.*"))`) or without one (`vis_regex(vis, NULL, 0)`). In the latter case, it gets the pattern from the search register / (which is what happens when pressing `n`). So it does use the pattern that is passed in.